File: Editorial/2001-2100/2087_Minimum_Cost_Homecoming_of_a_Robot_in_a_Grid/Go/1_Approach_Greedy.go
func minCost(startPos []int, homePos []int, rowCosts []int, colCosts []int) int {
    r1, c1 := startPos[0], startPos[1]
    r2, c2 := homePos[0], homePos[1]
    res := 0 // total cost
    
    // move to the row where the home is located, determine the direction of movement between rows, and calculate the corresponding cost
    if r2 >= r1 {
        for i := r1 + 1; i <= r2; i++ {
            res += rowCosts[i]
        }
    } else {
        for i := r2; i < r1; i++ {
            res += rowCosts[i]
        }
    }
    
    // move to the location of the house, determine the direction of movement between columns, and calculate the corresponding cost
    if c2 >= c1 {
        for i := c1 + 1; i <= c2; i++ {
            res += colCosts[i]
        }
    } else {
        for i := c2; i < c1; i++ {
            res += colCosts[i]
        }
    }
    
    return res
}

File: Editorial/2001-2100/2087_Minimum_Cost_Homecoming_of_a_Robot_in_a_Grid/Python3/1_Approach_Greedy.py
class Solution:
    def minCost(
        self,
        startPos: List[int],
        homePos: List[int],
        rowCosts: List[int],
        colCosts: List[int],
    ) -> int:
        r1, c1 = startPos[0], startPos[1]
        r2, c2 = homePos[0], homePos[1]
        res = 0  # total cost
        # move to the row where the home is located, determine the direction of movement between rows, and calculate the corresponding cost
        if r2 >= r1:
            for i in range(r1 + 1, r2 + 1):
                res += rowCosts[i]
        else:
            for i in range(r2, r1):
                res += rowCosts[i]
        # move to the location of the house, determine the direction of movement between columns, and calculate the corresponding cost
        if c2 >= c1:
            for i in range(c1 + 1, c2 + 1):
                res += colCosts[i]
        else:
            for i in range(c2, c1):
                res += colCosts[i]
        return res
